ra/ra.go

```diff
@@ -685,6 +714,42 @@ func (ra *RegistrationAuthorityImpl) checkAuthorizations(ctx context.Context, na
 			strings.Join(badNames, ", "),
 		)
 	}
+
+	return nil
+}
+
+func (ra *RegistrationAuthorityImpl) recheckCAA(ctx context.Context, names []string) error {
+	wg := sync.WaitGroup{}
+	ch := make(chan *probs.ProblemDetails, len(names))
+	for _, name := range names {
+		wg.Add(1)
+		go func(name string) {
+			resp, err := ra.caa.IsCAAValid(ctx, &vaPB.IsCAAValidRequest{
+				Domain: &name,
+			})
+			if err != nil {
+				ra.log.AuditErr(fmt.Sprintf("Rechecking CAA: %s", err))
+				ch <- probs.ServerInternal("Internal error rechecking CAA for " + name)
+			} else if resp.Problem != nil {
+				ch <- &probs.ProblemDetails{
+					Type:   probs.ProblemType(*resp.Problem.ProblemType),
+					Detail: *resp.Problem.Detail,
+				}
+			}
+			wg.Done()
+		}(name)
+	}
+	wg.Wait()
+	close(ch)
+	var fails []*probs.ProblemDetails
+	for err := range ch {
+		if err != nil {
+			fails = append(fails, err)
+		}
+	}
+	if len(fails) > 0 {
+		return fmt.Errorf("Rechecking CAA: %s", fails)
```

Comment:
Should this be a problem instead of an error? It seems like `ch` collects up `*probs.ProblemDetails` but then they're all stringified and smushed together here into a generic `error`. Are the string representations of the individual CAA lookup problems going to be meaningful?

Reply:
I think it makes sense to use a error here so a `probs.ProblemDetails` doesn't end up crossing the RPC boundary as a gRPC error. That said I think it should be a `berror`, probably `Unauthorized` or something, so it doesn't get converted to a `InternalServer` error at the WFE.